### DISCUSSION TITLE
How to create custom variants that is based on the html's data-attribute?

### DISCUSSION THREAD

**Discussion by goenning** [tailwindlabs/tailwindcss#9340] [Help] [answered]
Consider an html tag with a `data-os` attribute where possible values are `windows`, `linux` or `mac`.

I want to have variants so I can do things like: `mac:hidden` or `windows:mt-4` etc.

I tried using the addVariant like this: `addVariant('mac', 'html[data-os=mac]')`, but it complaints at build time as it needs to be an `at-rule` or `&:`. What other options do I have?

**Comment by adamwathan** [comment, marked as answer]
Hey! Should just need this:

`addVariant('mac', 'html[data-os=mac] &')`

**Comment by clintfisher** [comment]
@adamwathan  Is there any way to select a data attribute using an arbitrary variant?

`[[data-no-title]]:hidden`

**Reply by wongjn** [reply]
You're almost there! It is `[&[data-no-title]]:hidden`

**Reply by clintfisher** [reply]
The ampersand!  Thanks!
